openedx/core/djangoapps/oauth_dispatch/docs/README.rst

```diff
@@ -0,0 +1,102 @@
+    tokens.
+
+  * Its `Access Token View`_ returns JWTs as access tokens when a JWT token_type
+    is requested.
+
+  * It uses an edX custom JwtBuilder_ implementation to create the JWT.
+
+* The JwtBuilder_ uses the pyjwkest_ library for implementation of `JSON Web
+  Signature (JWS)`_ and other crypto to build and sign JWT tokens.
+
+.. _oauth_dispatch: https://github.com/edx/edx-platform/tree/master/openedx/core/djangoapps/oauth_dispatch
+.. _validator module: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/dot_overrides/validators.py
+.. _Access Token View: https://github.com/edx/edx-platform/blob/d21a09828072504bc97a2e05883c1241e3a35da9/openedx/core/djangoapps/oauth_dispatch/views.py#L89
+.. _JwtBuilder: https://github.com/edx/edx-platform/blob/d21a09828072504bc97a2e05883c1241e3a35da9/openedx/core/lib/token_utils.py#L15
+.. _pyjwkest: https://github.com/IdentityPython/pyjwkest
+.. _JSON Web Signature (JWS): https://tools.ietf.org/html/draft-ietf-jose-json-web-signature-41
+
+Clients & REST API Clients code
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* edX services, including LMS, use the edx-rest-api-client_ library
+  to make OAuth2 client requests and REST API calls.
+
+  * Built on top of slumber_, the edx-rest-api-client_ provides
+    a utility to retrieve an access token from the LMS. Its Auth_
+    classes create appropriate HTTP Authorization headers with
+    *Bearer* or *JWT* insertions as needed.
+
+  * It makes use of the PyJWT_ library for cryptographically creating
+    JWT tokens.
+    
+    * **Note:** Creation of JWT tokens in our system should only be done
+      by the OAuth Provider. This will break once we use *asymmetric* signing
+      keys, for which remote services will not have the private keys.
+
+.. _edx-rest-api-client: https://github.com/edx/edx-rest-api-client
+.. _slumber: https://github.com/samgiles/slumber
+.. _Auth: https://github.com/edx/edx-rest-api-client/blob/master/edx_rest_api_client/auth.py
+.. _PyJWT: https://github.com/jpadilla/pyjwt
+
+Authentication by REST endpoints
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Recently created edX REST endpoints use the `Django Rest Framework (DRF)`_.
+  The REST endpoint declares which type(s) of authentication it supports
+  or defaults to the *DEFAULT_AUTHENTICATION_CLASSES* value in DRF's 
+  *REST_FRAMEWORK* Django setting.
+
+* edX REST endpoints that support JWTs as access tokens declare the custom
+  edX JwtAuthentication_ class in its DRF authentication_classes_ scheme.
+
+  * JwtAuthentication_ is implemented in the edx-drf-extensions_ library.
+
+  * JwtAuthentication_ extends the JSONWebTokenAuthentication_ class
+    implemented in the django-rest-framework-jwt_ library.
+
+  * JwtAuthentication_ is used to authenticate an API request only
+    if it is listed in the endpoint's authentication_classes_ and the
+    request's Authorization header specifies "JWT" instead of "Bearer".
```

Comment:
Why do we break from the norm here and not use `Bearer`

Reply:
See my [response below](https://github.com/edx/edx-platform/pull/17266#discussion_r165818206).